datadog_checks_base/datadog_checks/checks/prometheus/mixins.py

```diff
@@ -217,6 +226,22 @@ def parse_metric_family(self, response):
             raise UnknownFormatError('Unsupported content-type provided: {}'.format(
                 response.headers['Content-Type']))
 
+    def _text_filter_input(self, input_gen):
+        """
+        Filters out the text input line by line to avoid parsing and processing
+        metrics we know we don't want to process. This only works on `text/plain`
+        payloads, and is an INTERNAL FEATURE implemented for the kubelet check
+        :param input_get: line generator
+        :output: generator of filtered lines
+        """
+        for line in input_gen:
+            for item in self._text_filter_blacklist:
```

Comment:
How large might this blacklist get? You might be able to achieve increased perf if you compile it to regex then a simple compiled_re.search.

`compiled_re = '|'.join(filters)`

Reply:
This is pretty drastic feature we should only use in case of obviously spammy content, not in place of the existing metric filtering. The kubelet check only needs one string, but I went with a list in case we'd need two or three (tops) in the future.